packages/datetime/src/dateRangePicker.tsx

```diff
@@ -98,6 +98,12 @@ export interface IDateRangePickerProps extends IDatePickerBaseProps, IProps {
      */
     onHoverChange?: (hoveredDates: DateRange, hoveredDay: Date, hoveredBoundary: Boundary) => void;
 
+    /**
+     * Called when the `shortcuts` props is enabled and the user changes the shortcut.
+     * When triggered, it will pass the selected `shortcut` and its index which is optional.
+     */
+    onShortcutChange?: (shortcut: IDateRangeShortcut, index?: number) => void;
```

Comment:
why is index optional? it seems like the Shortcuts component always passes it through the callback

Reply:
I think usually when a `shortcut` is changed user would like the details of the clicked `shortcut` rather than the `index`.
So it's optional.
If it is required then the user can pass the optional `index` parameter.
Any thoughts?

Reply:
but that's not how the typedef should be for the prop -- the library is _always_ passing `index`, so it shouldn't be marked as optional. a user can choose to assign a function which does or does not use the `index` argument, but they shouldn't have to deal with the case where `index` is possibly `undefined`.

if you make this line:
```suggestion
    onShortcutChange?: (shortcut: IDateRangeShortcut, index: number) => void;
```

then both of these are assignable as props by the user:
- `(shortcut: IDateRangeShortcut, index: number) => { ... }`
- `(shortcut: IDateRangeShortcut) => { ... }`

Reply:
oh ok.
I get it now.
Thanks.

Reply:
Done